packages/block-editor/src/store/actions.js

```diff
@@ -603,22 +603,76 @@ export const insertBlocks =
  *
  * @return {Object} Action object.
  */
-export function showInsertionPoint(
-	rootClientId,
-	index,
-	__unstableOptions = {}
-) {
-	const { __unstableWithInserter, operation, nearestSide } =
-		__unstableOptions;
-	return {
-		type: 'SHOW_INSERTION_POINT',
-		rootClientId,
-		index,
-		__unstableWithInserter,
-		operation,
-		nearestSide,
+export const showInsertionPoint =
+	( rootClientId, index, __unstableOptions = {} ) =>
+	( { select, dispatch } ) => {
+		const { __unstableWithInserter, operation, nearestSide, block } =
+			__unstableOptions;
+
+		let _destinationIndex = index;
+		let _destinationRootClientId = rootClientId;
+
+		if (
+			( block && block?.name ) ||
+			( Array.isArray( block ) && block?.length > 0 )
+		) {
+			const normalizedBlocks = Array.isArray( block ) ? block : [ block ];
+
+			const canInsertBlocks = () =>
+				normalizedBlocks.every( ( blockToInsert ) =>
+					select.canInsertBlockType(
+						blockToInsert.name,
+						_destinationRootClientId
+					)
+				);
+			if ( ! canInsertBlocks() && normalizedBlocks.length === 1 ) {
+				// If it's not possible to insert the block in the current location, try to find a new location up the tree
+				const rootBlocks = select.getBlocks();
```

Comment:
Why do we jump straight from the selected block to the root block instead of attempting to insert into parent blocks? Or am I missing something?